src/hivpy/resistance_mutations.py

```diff
@@ -0,0 +1,422 @@
+
+    def viral_load(self, pop: Population, sub_pop):
+        """
+        Update viral load for HIV+ individuals.
+        """
+        # get viral load outcomes
+        viral_load = pop.apply_function(self.calc_viral_load, 1, sub_pop)
+        pop.set_present_variable(col.VIRAL_LOAD, viral_load, sub_pop)
+
+    def calc_viral_load(self, person):
+        """
+        Returns an individual's viral load this time step.
+        Affected by number of active ART drugs, how long an individual has been on ART,
+        their ART adherence, as well as their viral load last time step.
+        """
+        # use resistance index to find the right (a, b, c) tuple
+        a, b, c = self.get_matrix_value(self.viral_load_matrix, person[col.RESISTANCE_INDEX])
+        # calculate base viral load value
+        # a * max_viral_load + b + c * min_vl_on_art
+        x = a * person[col.MAX_VIRAL_LOAD] + b + c * self.min_vl_on_art
+        # calculate viral load changes
+        viral_load = max(0, min(x + self.vl_stdev_on_art * rng.normal(), 6.5))
+
+        return viral_load
+
+    def cd4_change(self, pop: Population, sub_pop):
+        """
+        Update CD4 count for HIV+ individuals.
+        """
+        # FIXME: is there a better way to pass the the cd4_tm1 column string to calc_cd4_delta?
+        self.cd4_tm1_col = pop.get_correct_column(col.CD4, dt=1)
+        # get cd4 outcomes
+        cd4_outcomes = pop.apply_function(self.calc_cd4_delta, 1, sub_pop)
+        pop.set_present_variable(col.CD4, [i[0] for i in cd4_outcomes], sub_pop)
+        pop.set_present_variable(col.CD4_DELTA, [i[1] for i in cd4_outcomes], sub_pop)
+
+    def calc_cd4_delta(self, person):
+        """
+        Returns an individual's change in CD4 levels this time step.
+        Affected by age, sex, number of active ART drugs, how long an individual has been on ART,
+        their ART adherence, use of specific ART drugs, as well as CD4 levels last time step,
+        maximum CD4 levels, and individual rate of CD4 recovery on ART.
+        """
+        # use resistance index to lookup cd4 delta multiplier
+        x = self.get_matrix_value(self.cd4_delta_matrix, person[col.RESISTANCE_INDEX])
+
+        # find base cd4 recovery
+        base_cd4_recovery_on_art = 0
+        # recovery is hindered by a failing nnrti (or possibly insti) regimen
+        if (((person[col.ON_NEV] or person[col.ON_EFA]) or (self.failed_insti_hinders_cd4_recovery and person[col.ON_DOL]))
+                and not (person[col.ON_LPR] or person[col.ON_TAZ] or person[col.ON_DAR])
+                and person[col.NUM_ACTIVE_DRUGS] <= 2):
+            base_cd4_recovery_on_art = self.hindered_cd4_recovery
+        # recovery increases on pi
+        if person[col.ON_LPR] or person[col.ON_TAZ] or person[col.ON_DAR]:
+            base_cd4_recovery_on_art += self.cd4_recovery_pi_factor
+        # recovery decreases with age
+        base_cd4_recovery_on_art += (person[col.AGE] - 40) * - 0.3
```

Comment:
```suggestion
        base_cd4_recovery_on_art -= (person[col.AGE] - 40) *  0.3
```
I just think `-= ... *` is easier to read that `+= ... * -`